src/main/ExecutionService.ts

```diff
@@ -11,24 +14,34 @@ function isJsonString(s: string) {
 }
 
 export class ExecutionService {
-    executeCommands(pathToExecutable: string, commands: string[]): Promise<string> {
+    executeCommands(pathToExecutable: string, commands: string[]): Promise<CxCommandOutput> {
         return new Promise(function (resolve, reject) {
             let stderr = '';
+            let cxCommandOutput = new CxCommandOutput();
             const cp = spawn(pathToExecutable, commands);
             cp.stderr.on('data', function (chunk: string) {
                 stderr += chunk;
             });
             cp.on('error', reject)
                 .on('close', function (code: number) {
-                    if (code !== 0) {
-                        reject(stderr);
-                    }
+                    cxCommandOutput.exitCode = code;
+                    console.log("Exit code received from AST-CLI: " + code)
+                    resolve(cxCommandOutput);
+                    reject(stderr)
```

Comment:
can we remove the reject? we only want to output exit code, we don't need to reject it